### PR TITLE
Uncertainty estimation, calibration/sharpness, post-processing calibration

### PR DESCRIPTION
**Pull request by jirvin16**
**Problem**

We need to produce and evaluate uncertainty estimates around the CH4 predictions.

**Solution**

Implement uncertainty estimation using Laplace and Normal distributions, calibration/sharpness metrics, and post-processing calibration (uncertainty scaling).

**Tests**

```
python main.py test --sites MPM --models lasso,rf --predictors_paths predictors/mpm_predictors.txt --overwrite_results --split valid
python main.py test --sites MPM --models lasso,rf --predictors_paths predictors/mpm_predictors.txt --overwrite_results
python main.py test --sites NZKop --models lasso,rf --predictors_paths predictors/all.txt --overwrite_results --split valid
python main.py test --sites NZKop --models lasso,rf --predictors_paths predictors/all.txt --overwrite_results
```

### CODE REVIEW

**Comment by ylzhouchris** [comment]
**Problem**
Values of scaled sharpness (normalized or not) are too large (*10^8). Please see if this is expected. Maybe double check the implemented Platt scaling. 

Another optional minor is that it would be better to put all results together. The new pull request adds a "results" folder in the root directory. Results are stored in a "test.csv" which can be confusing to outputs from previous steps.

File: models/ensemble.py [hunk 1]
```diff
@@ -51,16 +53,76 @@ def feature_importances(self):
...
+        if distribution == "laplace":
+            numerator = np.absolute(truth.values - mean)
+            denominator = scale
+            return (numerator / denominator).mean()
+        elif distribution == "normal":
+            numerator = (truth.values - mean) ** 2
+            denominator = scale ** 2
+            return (numerator / denominator).mean()
+        else:
+            raise ValueError(f"Distribution {distribution} not supported.")
```
Comment (ylzhouchris) [inline models/ensemble.py:127]: Please double-confirm the implemented scaling of the distributions. It's leading to exceedingly large widths.mean() outputs.

**Review by ylzhouchris — CHANGES_REQUESTED** [review]

**Comment by jirvin16** [comment]
> **Problem**
> Values of scaled sharpness (normalized or not) are too large (*10^8). Please see if this is expected. Maybe double check the implemented Platt scaling.
> 
> Another optional minor is that it would be better to put all results together. The new pull request adds a "results" folder in the root directory. Results are stored in a "test.csv" which can be confusing to outputs from previous steps.

You were absolutely right re: large sharpness! There was a bug with the Laplace implementation that led to exponentiated values. That's been fixed and the sharpness values look much more reasonable. Nice catch.

Thanks for raising the results issue too. I've tried to clean this up by writing results for each model to the corresponding model directory, and writing all aggregated results to `results/` - let me know what you think.

**Comment by ylzhouchris** [comment]
Hi Jeremy, thanks for the revisions made! I think the pull request can be approved.

**Review by ylzhouchris — APPROVED** [review]

**Pull request merged**